src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -0,0 +1,446 @@
+                        return;
+                    }
+
+                    foreach (var platformSpecificOperation in platformSpecificOperations)
+                    {
+                        var value = analysisResult[platformSpecificOperation.Key.Kind, platformSpecificOperation.Key.Syntax];
+                        PlatformAttributeInfo attribute = platformSpecificOperation.Value.FirstOrDefault();
+
+                        if (value.Kind == GlobalFlowStateAnalysisValueSetKind.Unknown)
+                        {
+                            if (platformSpecificOperation.Key.TryGetContainingLocalOrLambdaFunctionSymbol(out var containingSymbol))
+                            {
+                                var localResult = analysisResult.TryGetInterproceduralResultByDefinition(containingSymbol);
+                                if (localResult != null)
+                                {
+                                    var localValue = localResult[platformSpecificOperation.Key.Kind, platformSpecificOperation.Key.Syntax];
+                                    if (localValue.Kind == GlobalFlowStateAnalysisValueSetKind.Known)
+                                    {
+                                        if (IsKnownValueGuarded(attribute, localValue))
+                                        {
+                                            continue;
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                        else if (value.Kind == GlobalFlowStateAnalysisValueSetKind.Known)
+                        {
+                            if (IsKnownValueGuarded(attribute, value))
+                            {
+                                continue;
+                            }
+                        }
+
+                        ReportDiagnostics(platformSpecificOperation.Key, attribute, context);
+                    }
+                }
+                finally
+                {
+                    platformSpecificOperations.Free();
+                }
+
+                return;
+
+                OperationVisitor CreateOperationVisitor(GlobalFlowStateAnalysisContext context) => new OperationVisitor(guardMethods, osPlatformType, context);
+            });
+        }
+
+        private static bool IsKnownValueGuarded(PlatformAttributeInfo attribute, GlobalFlowStateAnalysisValueSet value)
+        {
+            foreach (var analysisValue in value.AnalysisValues)
+            {
+                if (analysisValue is RuntimeMethodValue info)
+                {
+                    if (!info.Negated)
+                    {
+                        if (IsOSPlatformsEqual(attribute.PlatformName, info.PlatformPropertyName))
+                        {
+                            if (info.InvokedPlatformCheckMethodName == s_platformCheckMethodNames[0])
```

Comment:
I think it is better to write this as a switch case, with explicit names used in each case clause and an `NotImplementedException` thrown for default/unknown `InvokedPlatformCheckMethodName`. This should ensure that changing the order or type of `s_platformCheckMethodNames` does not impact this code, and if more methods are added to it in future, we get the not implemented exception and add handling for it.